### ISSUE TITLE
isExternal logic was removed since v1 and causes issues with other plugins

### ISSUE THREAD

**Issue by bdwain** [sebastian-software/rollup-plugin-rebase#18]
I think #14 is because the rebase plugin used to expose an isExternal method that could be used in other plugins or rollup itself

```js
const rebase = rebasePlugin({ outputFolder, input })

rollup({
  input: 'foo.js',
  output: {format: 'es', file: 'bar.js'},
  plugins: [
    rebase,
    includePaths({
      paths: ['src/'],
      external: file => {
        if(rebase.isExternal(file)){
          return true;
        }
        if(file[0] === '/' || file[0] === '.'){
          return false;
        }
        const includedFolders = ['js', 'img', 'src'];
        return includedFolders.every(f => !file.startsWith(f + '/'));
      }
    })
  ]
})
```

Is there a reason that was deleted? It's holding me back from upgrading to v3 from v1.

**Comment by swernerx** [comment]
Thanks for the issue report. Actually, the method was meant to be used internally only. As this is not required anymore in the recent version, it was removed for cleaning up the API surface. I'll check whether I can offer anything comparable, but would probably name it differently e.g. `isAsset()`

**Comment by bdwain** [comment]
cool thank you! that would be much appreciated!

**Comment by swernerx** [comment]
I was a little unsure about what exactly is required in your set-up. For now I added three new methods which allow read-only access to internal data structure.

Whether the given ID is a wrapper file generated by this plugin:
- `isRebaseWrapper(wrapperId)`

Whether the given assetId is an registered asset file:
- `isRebaseAsset(assetId)`

Whether the given source is was resolved by this plugin:
- `isRebaseSource(fileSource)`

**Comment by swernerx** [comment]
Please re-open with specific details/use-case when the issue persists.

**Comment by bdwain** [comment]
Hi sorry I didn't get back to you sooner. I tried this and had trouble getting it to work. I'm not exactly sure which of the 3 new methods I need, but I think the error might actually be something different.

Basically what I'm looking for is a way to tell another plugin that a file has already been rebased. I use the rollup include paths plugin to allow an import like this

```js
import img from 'img/foo.png';
//instead of needing something like
import img from '../../../../img/foo.png';
```

However, since rebase is moving these files and changing the import to relative, I need it to tell the other plugin to ignore that file somehow. 

I tried doing something like this in the isExternal function of the other plugin to replace the old call to isExternal (see my original post)

```js
if(rebase.isRebaseWrapper(file) || rebase.isRebaseAsset(file) || rebase.isRebaseSource(file)){
  return true;
}
```

and it didn't work. I was hoping for a function I could just replace the old isExternal with.

But now I think the errors i'm getting are unrelated anyway. I get an error in the rebase plugin saying something like

```
ENOENT: no such file or directory, open '/Users/bwain/dev/my-project/src/js/components/foo/bar/img/img-folder/img-name.png'. 
```

So it seems to be treating the `import img from 'img/img-folder/img-name.png'` like a relative import? In the past (v1 of rebase plugin) it would handle these correctly, so I'm not sure what changed.
